src/SfxWeb/src/app/modules/event-store/event-store/event-store.component.ts

```diff
@@ -274,13 +281,25 @@ export class EventStoreComponent implements OnInit, OnDestroy {
                                     if (!inputEvent.related) {
                                         inputEvent.related = [];
                                     }
-                                    if(Utils.result(iterEvent, relevantEventType.result)) {
-                                        action = "Action: " + Utils.result(iterEvent, relevantEventType.result) + "<br/><br/>";
-                                    }
-                                    iterEvent.reasonForEvent = action;
                                     inputEvent.related.push(iterEvent);
                                     addedEvents.push(iterEvent);
                                 }
+                            } else if(relevantEventType.eventType == "self") {
+                                let propMaps = true;
```

Comment:
lets move this logic into another function and pass in all that is relevent.

we could then check inside if the source and target are the same and if so set the target to the property, this should cut down on duplicate code and we could look at adding a pass where it does all self reference checks.

Reply:
there isn't much duplicate code besides for this `if (sourceVal != null && sourceVal != undefined && targetVal != null && targetVal != undefined && sourceVal != targetVal) {
                                        propMaps = false;
                                    }`